src/pages/Devices.tsx

```diff
@@ -0,0 +1,608 @@
+import { Box,  Grid, CardContent, Typography, SelectChangeEvent, SpeedDial, SpeedDialAction, SpeedDialIcon,  } from '@mui/material';
+import RowContainerBetween from '../components/shared/RowContainerBetween';
+import { Add } from '@mui/icons-material';
+import { DEFAULT_COLORS } from '../constants';
+import { Link, useNavigate, useOutletContext, } from 'react-router-dom';
+import React, { useContext, useEffect, useState } from 'react';
+import { type Device } from 'waziup';
+import CreateDeviceModalWindow from '../components/ui/ModalCreateDevice';
+import EditDeviceModal from '../components/ui/EditDeviceModal';
+import { DevicesContext, SensorX } from '../context/devices.context';
+import { devEUIGenerateFc, differenceInMinutes, lineClamp,  } from '../utils';
+import PrimaryIconButton from '../components/shared/PrimaryIconButton';
+import SensorActuatorInfo from '../components/shared/SensorActuatorInfo';
+import MenuComponent from '../components/shared/MenuDropDown';
+import RowContainerNormal from '../components/shared/RowContainerNormal';
+const initialNewDevice: Device = {
+    actuators: [],
+    created: new Date(),
+    id: '',
+    meta: {
+        type: '',
+        codec: '',
+        lorawan: null,
+    },
+    modified: new Date(),
+    name: '',
+    sensors: [],
+}
+import DeviceImage from '../assets/device.png';
+function Devices() {
+    const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+    const open = Boolean(anchorEl);
+    const navigate = useNavigate();
+    const { devices, wazigateId, getDevicesFc,sortDevices,showDialog } = useContext(DevicesContext);
+    const [selectedDevice, setSelectedDevice] = useState<null | Device>(null);
+    const [newDevice, setNewDevice] = useState<Device>(initialNewDevice);
+    const [matches] = useOutletContext<[matches: boolean,matchesMd: boolean]>();
+    const handleToggleModal = () => {
+        setOpenModal(!openModal);
+        if (!openModal) {
+            setNewDevice(initialNewDevice);
+        }
+    };
+    const [openModal, setOpenModal] = useState(false);
+    const [openEditModal, setOpenEditModal] = useState(false);
+    const handleToggleEditModal = () => {
+        setOpenEditModal(!openEditModal);
+    }
+    useEffect(()=>{
+        if (devices.length===0) {
+            getDevicesFc();
+        }
+        window.wazigate.subscribe('devices', (dt,tpc)=>{
```

Comment:
Let's remove the MQTT subcriptions at the moment. We can do it in the next version.